iothub/device/src/DirectMethod/EdgeModuleDirectMethodRequest.cs

```diff
@@ -80,7 +80,7 @@ public EdgeModuleDirectMethodRequest(string methodName, object payload)
         /// The direct method payload.
         /// </summary>
         [JsonProperty("payload", NullValueHandling = NullValueHandling.Include)]
-        internal object Payload { get; }
+        internal byte[] Payload { get; }
```

Comment:
This field will be deserialized directly when using module-module method invocations over HTTP. Make sure that scenario is tested to ensure serialization works as expected. The long haul tests should cover this scenario.